### DISCUSSION TITLE
the best way to hide a row with a value and show it again

### DISCUSSION THREAD

**Discussion by algospa** [TanStack/table#5314] [Q&A]
Hi need help, im stack all day with tryin to hide a row with a value from the status column with 'D' as a string with no luck, i tried all online but nothing worked with tanstack table library functions. Im even trying to gett a second working endpoint from my api in order to see if i can get a work around but no luck, someone have an idea?

**Comment by goatfryed** [comment]
There are various ways to do this depending on your requirement. It might actually be most useful to just use plain old js Array.filter before you pass the data to useTable or when you map your rows to a display.

If you want to have a more library centric solution, I'd consider to add a virtual (non-displayed) search column with a custom filter function and set a fitting initial filter. Consider the example and documentation on [filters](https://tanstack.com/table/v8/docs/framework/react/examples/filters) and also discussion #5328 

Difficult to say without more context, but I'd highly consider to run a data.filter(...) before you pass the data to the useTable, if this is never relevant and basically API overhead.
